cypress/support/commands.js

```diff
@@ -53,14 +53,12 @@ Cypress.Commands.add('testSetup', () => {
  * @returns {Cypress.Chainable<Cypress.Response>} A cypress request for signing in a user.
  */
 Cypress.Commands.add('loginUser', ({ email, password }) => {
-  const encodedEmail = encodeURIComponent(email);
-  const encodedPassword = encodeURIComponent(password);
-
-  return cy.request(
-    'POST',
-    '/users/sign_in',
-    `utf8=%E2%9C%93&user%5Bemail%5D=${encodedEmail}&user%5Bpassword%5D=${encodedPassword}&user%5Bremember_me%5D=0&user%5Bremember_me%5D=1&commit=Continue`,
-  );
+  return cy.visit('/enter').then(() => {
```

Comment:
This seems sensible to me but probably takes a bit longer since it evolves actual page navigation and DOM interactions. Is there a concrete problem with the existing implementation you're trying to solve?

Reply:
This was an experiment I came up with while we were investigating why some tests were flaking last week. The problem we were dealing with was:

- `user1` signs in
- Some work is done
- `user1` signs out
- `user2` signs in
- **problem:** `user1` is still signed in

My train of thought was: 

> If we run the authentication process via Cypress instead of a one-off POST request, then the built-in "waits" (driver waiting for the page to load correctly) would help avoid these mismatches. [This thread](https://forem-team.slack.com/archives/C012YSX5ZC1/p1623961612024000?thread_ts=1623955953.018500&cid=C012YSX5ZC1) has more context.

This could be entirely unnecessary if the tests stop flaking or if we find an alternative to sort out this authentication problem. I'm totally okay if we consider this PR is not needed 🙂

Reply:
I guess I just wasn't aware of the Slack thread's context (though it does seem familiar). I'm all for reducing the flakiness of the Cypress tests, I've been spending quite _a lot_ of time rebuilding lately.